### PR TITLE
test(esl-animate): cover esl-animate module parts with tests and code improvements

### PR DESCRIPTION
**Pull request by fshovchko**
Tests are covering the following areas of module:
ESL Animate Service smoke (basic + options)
ESL Animate Element unit (wrapper) tests
ESL Animate Mixin unit (wrapper) tests
#1280

### CODE REVIEW

File: src/modules/esl-animate/test/intersectionObserver.mock.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  public takeRecords(): any {}
+
+  @bind
+  private _onObserve(e: any): void {
+    const {intersectionRatio, isIntersecting} = e.detail;
+    this.callback([{intersectionRatio, target: e.target, isIntersecting}] as IntersectionObserverEntry[], this);
+  }
+
+  public observe(element: Element): void {
+    element.addEventListener('intersection', this._onObserve);
```
Comment (ala-n) [inline src/modules/esl-animate/test/intersectionObserver.mock.ts:33]: Awesome idea 👍🏻

File: src/modules/esl-animate/test/animate.mixin.test.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+jest.mock('../../esl-utils/dom/ready', () => ({
+  onDocumentReady: (cb: any) => cb()
+}));
+
+describe('ESLAnimate', () => {
+  const $el = document.createElement('div');
+  $el.toggleAttribute(ESLAnimateMixin.is, true);
+  document.body.appendChild($el);
+
+  window.IntersectionObserver = MockIntersectionObserver;
```
Comment (ala-n) [inline src/modules/esl-animate/test/animate.mixin.test.ts:14]: jest.spyOn(window, 'IntersectionObserver').mockImplementation(MockIntersectionObserver);

**Review by ala-n — COMMENTED** [review]
Proposing to simplify test flow:
We are totally sure that `AnimateMixin` and `AnimateElement` are just auxiliary sugar above Animate Service. Keeping that in mind, we can choose another strategy to test the module:
- `animate.mixin`, `animate.element` - total unit tests that considers `AnimationService` as a separate 100% tested unit (so the result tests mock the service and checks correct registration but not the whole animate cases)
  PS: smoke with integration test is applicable too but there is no reasons no make whole test list of integration tests
- `animate.service` - is the exact target for all test cases regarding animate module behavior

File: src/modules/esl-animate/test/animate.mixin.test.ts (hunk 2 above)
Comment (fshovchko) [reply src/modules/esl-animate/test/animate.mixin.test.ts]: Cannot spy on undefined property. Using jest.fn() instead

**Review by dshovchko — APPROVED** [review]

**Review by ala-n — APPROVED** [review]

**Review by yadamskaya — APPROVED** [review]

**Review by NastaLeo — APPROVED** [review]

**Pull request merged**